File: runme/mcgen/local/datan/exasrc/e2test.c
#include <stdio.h>
#include <math.h>
#include <string.h>
#include "datsrc.h"

main(void)
{
    /* System generated locals */
    integer i__1, i__2;

    /* Local variables */
    static integer i, n, iexp, nexp, nf;
    static double dels, q, r[1000], s, t, sigma, xmean, delxm, x0, dels2, c,
                  confid, al0, ssq;
    static char text[15];
/* identify program to user */
    printf("%s\n", "Program E2TEST simulates Gaussian sample");
    printf("%s\n\n", "and performs t-test on mean.");
/* ask for input */
    printf("%s\n", "Enter number NEXP of experiments (>0):");
    printf("%s", "> ");
    scanf("%li", &nexp);
    printf("%s\n", "Enter number N of elements in sample (>0):");
    printf("%s", "> ");
    scanf("%li", &n);
    printf("%s\n", "Enter mean X0 for sample:");
    printf("%s", "> ");
    scanf("%lf", &x0);
    printf("%s\n", "Enter standard deviation SIGMA for sample (>0.):");
    printf("%s", "> ");
    scanf("%lf", &sigma);
    printf("%s\n", "Enter hypothesis AL0 for mean:");
    printf("%s", "> ");
    scanf("%lf", &al0);
    printf("%s\n", "Enter confidence limit CONFID (0.<CONFID<1.):");
    printf("%s", "> ");
    scanf("%lf", &confid);
    c = (confid + 1.) * .5;
    nf = n - 1;
    q = sqstud_(&c, &nf);
/* write heading for table */
    printf("%s\n", "IEXP    N       XMEAN       t          Q");
/* loop over all experiments */
    i__1 = nexp;
    for (iexp = 1; iexp <= i__1; ++iexp) {
/* simulate sample R */
	rnstnr_(r, &n);
	i__2 = n;
	for (i = 1; i <= i__2; ++i) {
	    r[i - 1] = x0 + r[i - 1] * sigma;
	}
/* compute sample mean and variance and t */
	smmnvr_(r, &n, &xmean, &delxm, &ssq, &dels2, &s, &dels);
	t = (xmean - al0) * sqrt((double) n / ssq);
	if (abs(t) > q) {
	    strcpy(text, " t-test failed");
	} else {
	    strcpy(text, "              ");
	}
/* output */
        printf("%5li%5li%12.5lf%12.5lf%12.5lf%s\n",
               iexp, n, xmean, t, q, text);
    }
    return 0;
} /* main */

File: runme/mcgen/local/datan/exasrc/e2test.f
      PROGRAM E2TEST
      IMPLICIT DOUBLE PRECISION(A-H,O-Z)
      DIMENSION R(1000)
      CHARACTER*15 TEXT
C identify program to user
      WRITE(*,*)' Program E2TEST simulates Gaussian sample'
      WRITE(*,*)' and performs t-test on mean'
      WRITE(*,*)' '
C ask for input
      WRITE(*,*)' Enter number NEPX of experiments (>0)'
      WRITE(*,*)' >'
      READ(*,*)NEXP
      WRITE(*,*)' Enter number N of elements in sample (>0)'
      WRITE(*,*)' >'
      READ(*,*)N
      WRITE(*,*)' Enter mean X0 for sample'
      WRITE(*,*)' >'
      READ(*,*)X0
      WRITE(*,*)' Enter standard deviation SIGMA for sample (>0.)'
      WRITE(*,*)' >'
      READ(*,*)SIGMA
      WRITE(*,*)' Enter hypothesis AL0 for mean'
      WRITE(*,*)' >'
      READ(*,*)AL0
      WRITE(*,*)' Enter confidence limit CONFID (0.<CONFID<1.)'
      WRITE(*,*)' >'
      READ(*,*)CONFID
      C=0.5D0*(1.D0+CONFID)
      NF=N-1
      Q=SQSTUD(C,NF)
C write heading for table
      WRITE(*,*)' IEXP    N       XMEAN       t          Q'
C loop over all experiments
      DO 30 IEXP=1,NEXP
C simulate sample R
        CALL RNSTNR(R,N)
        DO 10 I=1,N
          R(I)=X0+R(I)*SIGMA
10      CONTINUE
C compute sample mean and variance and t
        CALL SMMNVR(R,N,XMEAN,DELXM,SSQ,DELS2,S,DELS)
        T=(XMEAN-AL0)*SQRT(DBLE(N)/SSQ)
        IF(ABS(T).GT.Q) THEN
          TEXT=' t-test failed'
        ELSE
          TEXT=' '
        END IF
C output
        WRITE(*,'(1X,2I5,3F12.5,A)')IEXP,N,XMEAN,T,Q,TEXT
30    CONTINUE
      END
